pykafka/producer.py

```diff
@@ -18,16 +18,17 @@
 limitations under the License.
 """
 __all__ = ["Producer", "AsyncProducer"]
+from collections import defaultdict, deque, namedtuple
 import itertools
 import logging
+import threading
```

Comment:
This poses an issue since we also need to use `Condition` variables here. I'm not sure how to integrate those into the handler.